[Docs] add docs for signing osquery installers with fleetctl (#3577)
fleetdm/fleet: docs/01-Using-Fleet/04-Adding-hosts.md

When you install the generated osquery installer on a host, this host will be automatically enrolled in the specified Fleet instance.

> ## Signing installers

Reviewer: ```suggestion
### Signing installers
```
Can you please add a third `#` character here? This is because I think we'd like the "Signing installers" section to look like a subsection of the "Osquery installers" section.

>   >**Note:** Windows signing is not yet supported within `fleetctl` (working on that this week), but the resulting MSI could be signed after building.

Reviewer: ```suggestion
  >**Note:** Currently, the `fleetctl package` command does not provide support for signing and notarizing Windows osquery installers. Windows installers can be signed after building.
```

Reviewer: @DominusKelvin can you please move the "Note" section to the bottom of the "Signing installers" section. This is because the information in the "Note" section is usually the least valuable information for the user. 

If you have the time, can you update the "Documentation" section in the `handbook/product.md` file to information contributors to place the "Note" subsection at the end of its respective section. The greyed out "Note" subsection is near the bottom of other sections in the docs. 

I also suggest making the Windows sentence consistent with the following macOS sentence:

> The `fleetctl package` command provides support for signing and notarizing macOS osquery installers via the
`--sign-identity` and `--notarize` flags.

Author: > Suggested change 
>   >**Note:** Windows signing is not yet supported within `fleetctl` (working on that this week), but the resulting MSI could be signed after building.
>   >**Note:** Currently, the `fleetctl package` command does not provide support for signing and notarizing Windows osquery installers. Windows installers can be signed after building.

Hey Noah. I might be wrong but I don't think the term "Notarizing" applies to Windows app. I think it's more of a Mac thing.

Reviewer: > I might be wrong but I don't think the term "Notarizing" applies to Windows app. I think it's more of a Mac thing.

@DominusKelvin I think you're right. I don't know this. Good catch!

Author: So I left out "notarising" and applied the rest of the changes.

> The `fleetctl package` command provides suppport for signing and notarizing macOS osquery packages via the

Reviewer: ```suggestion
The `fleetctl package` command provides suppport for signing and notarizing macOS osquery installers via the
```

I propose updating "packages" to "installers" here. In the documentation, I'd like to consistently refer to the packages as "Fleet's osquery installers," "macOS installers," "macOS osquery installers," etc.

this, modify the command to create an osquery installer to look with the `--sign-identity` and

> `--notarize` flags like the example below:

Reviewer: ```suggestion
The `fleetctl package` command provides suppport for signing and notarizing macOS osquery packages via the 
`--sign-identity` and `--notarize` flags. 

Check out the example below:

Reviewer: I propose combining the 2 sentences into 1 here. This way, the documentation is as brief as possible while still delivering the user clear instructions.
